### PR TITLE
Add Time, Datepicker, LanguagePickerSelect, and ListSort to sandbox

### PR DESCRIPTION
**Pull request by brianfleming**
## Description

While working on a couple recent PRs and preparing designs for shared feeds, I noticed some small issues with a few CDS components as well as them not being in the sandbox. This PR adds
- Time
- Datepicker
- LanguagePickerSelect
- ListSort
- Updates ToggleButtonGroup component to add in variants (size) and some functionality we skipped during initial development of the component

to the sandbox and fixes some style issues with disabled states and labeling

## Type of change

- [x] Performance improvement and/or refactoring (non-breaking change that keeps existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Security mitigation or enhancement
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Automated test (add or update automated tests)

## How has this been tested?

sandbox

### Datepicker
Screenshot 2023-11-14 at 11 12 01 AM

### LanguagePickerSelect
Screenshot 2023-11-14 at 11 12 05 AM

### ListSort
Screenshot 2023-11-14 at 11 12 11 AM

### Time
Screenshot 2023-11-14 at 11 11 58 AM

### ToggleButtonGroup
Screenshot 2023-11-15 at 9 38 57 AM

## Checklist

- [x] I have performed a self-review of my own code
- [x] I've made sure my branch is runnable and given good testing steps in the PR description
- [x] I considered secure coding practices when writing this code. Any security concerns are noted above.
- [x] I have commented my code in hard-to-understand areas, if any
- [ ] I have made needed changes to the README
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If I implemented any new components, they are self-contained, their `propTypes` are declared and they use React Hooks and, if data-fetching is required, they use Relay Modern with fragment containers
- [ ] If my components involve user interaction - specifically button, text fields, or other inputs - I have added a [BEM-like class name](https://meedan.atlassian.net/wiki/spaces/ENG/pages/1327628289/Naming+conventions+for+interactive+elements) to the element that is interacted with
- [x] To the best of my knowledge, any new styles are applied according to the design system
- [ ] If I added a new external dependency, I included a rationale for doing so and an estimate of the change in bundle size (e.g., checked in https://bundlephobia.com/)

### CODE REVIEW

File: src/app/components/cds/forms/LanguagePickerSelect.js [hunk 1]
```diff
@@ -97,6 +100,7 @@ LanguagePickerSelect.propTypes = {
   languages: PropTypes.arrayOf(PropTypes.string),
   isDisabled: PropTypes.bool,
   showLabel: PropTypes.bool,
+  helpContent: PropTypes.element,
```
Comment (brianfleming) [inline src/app/components/cds/forms/LanguagePickerSelect.js:103]: Added helpcontent to this component to use in some upcoming help text changes from Noah

File: src/app/components/team/Languages/AddLanguageAction.js [hunk 2]
```diff
@@ -143,6 +143,7 @@ const AddLanguageAction = ({ team }) => {
                           description="Label to language selection dropdown"
                         />
                       }
+                      helpContent={<FormattedMessage id="addLanguageAction.selectLanguageHelp" defaultMessage="After adding this language, be sure to customize the chatbot’s responses in the tipline settings." description="Help text for next steps after selecting a language" />}
```
Comment (brianfleming) [inline src/app/components/team/Languages/AddLanguageAction.js:146]: This was help text that somehow I forgot to include in https://github.com/meedan/check-web/pull/1722

File: src/app/components/media/AutoCompleteMediaItem.js [hunk 3]
```diff
@@ -273,17 +273,20 @@ const AutoCompleteMediaItem = (props, context) => {
...
-                      <FormattedMessage id="autoCompleteMediaItem.notFound" defaultMessage="No matches found" description="Status message when a search returned no results" /> : null }
-                    { searchResult.error ?
-                      <FormattedMessage id="autoCompleteMediaItem.error" defaultMessage="Sorry, an error occurred while searching. Please try again and contact {supportEmail} if the condition persists." description="Status message when a search resulted in an error" values={{ supportEmail: stringHelper('SUPPORT_EMAIL') }} /> : null }
-                    { searchResult.items && searchResult.items.length > 0 ?
-                      <FormattedMessage id="autoCompleteMediaItem.results" defaultMessage="{count, plural, one {# result} other {# results}}" description="Count of search results" values={{ count: searchResult.items.length }} /> : null }
-                  </React.Fragment> : <FormattedMessage id="autoCompleteMediaItem.type" defaultMessage="Type above to search" description="Helper text displayed next to search field to look for items to be imported" />
+                <span id="autocomplete-media-item-helper-text">
```
Comment (brianfleming) [inline src/app/components/media/AutoCompleteMediaItem.js:276]: I think an integration test is looking for this and it was removed at some point

File: src/app/components/cds/inputs/LanguagePickerSelect.js [hunk 4]
```diff
@@ -70,8 +71,9 @@ const LanguagePickerSelect = ({
                 <TextField
                   iconLeft={<LanguageIcon />}
                   iconRight={<ChevronDownIcon />}
-                  label={showLabel ? <FormattedMessage id="LanguagePickerSelect.selectLanguageLabel" defaultMessage="Language" description="Label for input to select language of fact-check" /> : null}
+                  label={label && <FormattedMessage id="LanguagePickerSelect.selectLanguageLabel" defaultMessage="Language" description="Label for input to select language of fact-check" />}
```
Comment (amoedoamorim) [inline src/app/components/cds/inputs/LanguagePickerSelect.js:74]: I guess this should be `label || <Formatt...` if you want `label` to display if set

File: src/app/components/cds/inputs/ToggleButtonGroup.js [hunk 5]
```diff
@@ -56,15 +69,17 @@ const ToggleButtonGroup = ({
...
   label: '',
   variant: 'outlined',
 };
 
 ToggleButtonGroup.propTypes = {
   className: PropTypes.string,
+  size: PropTypes.oneOf(['default', 'small', 'large']),
   helpContent: PropTypes.element,
-  label: PropTypes.string,
+  label: PropTypes.element,
```
Comment (amoedoamorim) [inline src/app/components/cds/inputs/ToggleButtonGroup.js:82]: I think it's better to have label as PropTypes.node as it accounts for strings, number and elements, anything that can be rendered

**Review by amoedoamorim — APPROVED** [review]

File: src/app/components/cds/inputs/ToggleButtonGroup.js (hunk 5 above)
Comment (brianfleming) [reply src/app/components/cds/inputs/ToggleButtonGroup.js]: Thanks @amoedoamorim I changed the new proptypes in this PR to be .node instead

**Review by amoedoamorim — APPROVED** [review]
:100:

**Pull request merged**